typing/includemod_errorprinter.ml

```
@@ -172,19 +178,60 @@ module Illegal_permutation = struct
       (Includemod.kind_of_field_desc kind)
       Style.inline_code (Ident.name id)
 
-  let pp ctx_printer env ppf (mty,c) =
+  let illegal_permutation ctx_printer env ppf (mty,c) =
+    match first_change c with
+    | None | Some (_, (Primitive_coercion _ | Alias_coercion _)) ->
```

The behaviour with `None` is changing a bit. The previous code had a comment saying "this should not happen", and there is some logic behind that (if we get an illegal permutation error we should not have an identity coercion), so I consider it's fine.
It does rely on identity coercions being systematically collapsed into `Tcoerce_none`, but I believe this is the case.

---

Indeed, this my usual hesitation between writing default error message for impossible case or failing early.